{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Hypothetical HBV Clinical Data Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: In order to mimick the process of data processing used in the study, I have generated hypothetical HBV clinical data. The data shown below has no relation to the original clinical data. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Generate a random dataframe for Sample (i.e. unique patients' number used during the clinical study). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the random dataframe for Sample (S)\n",
    "dataframe_sample = pd.DataFrame(np.random.randint(1,\n",
    "                                                  100, size=(1000, 1)),\n",
    "                                columns=list('S'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Generate a random dataframe for Reference Genome (i.e. genotype-specific reference sequence (GenBank accession numbers: AF090842, AB033554, AB033556, AF121240, and AB032431, for genotypes A to E, respectively))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Reference Genome used in the clinical study\n",
    "GenBank_sequences = [\"AF090842\",\n",
    "                     \"AB033554\",\n",
    "                     \"AB033556\",\n",
    "                     \"AF121240\",\n",
    "                     \"AB032431\"]\n",
    "\n",
    "# Create the dataframe for Reference Genome (R)\n",
    "dataframe_reference_genome = pd.DataFrame(np.random.choice\n",
    "                                          (GenBank_sequences,\n",
    "                                           size=1000,\n",
    "                                           replace=True,\n",
    "                                           p=None),\n",
    "                                          columns=list('R'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 3: Generate a random dataframe for Gene (i.e. genes encoded by the genome: C (PreCore, Core), P (Polymerase, RT), S (PreS1, PreS2, HBsAg), X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Genes used in the clinical study\n",
    "gene_genome = [\"PreCore\", \"Core\",\n",
    "               \"Polymerase\", \"RT\",\n",
    "               \"PreS1\", \"PreS2\",\n",
    "               \"HBsAg\", \"X\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the dataframe for Genes (G)\n",
    "dataframe_gene_genome = pd.DataFrame(np.random.choice\n",
    "                                     (gene_genome, size=1000,\n",
    "                                      replace=True, p=None),\n",
    "                                     columns=list('G'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 4: Generate a random dataframe for Effect (i.e. intragenic or missense variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Effect used in the clinical study\n",
    "effect_variant = [\"intragenic_variant\", \"missense_variant\"]\n",
    "\n",
    "# Create the dataframe for Effect (E)\n",
    "dataframe_effect = pd.DataFrame(np.random.choice(effect_variant,\n",
    "                                                 size=1000,\n",
    "                                                 replace=True,\n",
    "                                                 p=None), columns=list('E'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 5: Generate a random dataframe for amino acid variant (e.g. p.Pro156Ser). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for p.amino_acid (e.g. p.Pro in p.Pro156Ser)\n",
    "p_amino_acid = [\"p.Ala\", \"p.Arg\", \"p.Asn\", \"p.Asp\", \"p.Cys\",\n",
    "                \"p.Gln\", \"p.Glu\", \"p.Gly\", \"p.His\", \"p.Ile\", \"p.Leu\",\n",
    "                \"p.Lys\", \"p.Met\", \"p.Phe\", \"p.Pro\", \"p.Ser\",\n",
    "                \"p.Thr\", \"p.Trp\", \"p.Tyr\", \"p.Val\", \"p.Asx\", \"p.Glx\"]\n",
    "\n",
    "# Generate the random p.amino_acid\n",
    "random_p_amino_acid = [random.choice(p_amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate the random amino acid position\n",
    "random_digit_amino_acid_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "# Convert list of interger to list of string\n",
    "random_digit_amino_acid = list()\n",
    "for element in random_digit_amino_acid_int:\n",
    "    random_digit_amino_acid.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for amino_acid (e.g. Ser in p.Pro156Ser)\n",
    "amino_acid = [\"Ala\", \"Arg\", \"Asn\", \"Asp\", \"Cys\",\n",
    "              \"Gln\", \"Glu\", \"Gly\", \"His\", \"Ile\", \"Leu\",\n",
    "              \"Lys\", \"Met\", \"Phe\", \"Pro\", \"Ser\",\n",
    "              \"Thr\", \"Trp\", \"Tyr\", \"Val\", \"Asx\", \"Glx\"]\n",
    "\n",
    "# Generate the random amino_acid\n",
    "random_amino_acid = [random.choice(amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine p.amino_acid, position #\n",
    "# and amino_acid into one string for each index\n",
    "amino_acid_variant = map(lambda a, b, c: a + b + c,\n",
    "                         random_p_amino_acid,\n",
    "                         random_digit_amino_acid,\n",
    "                         random_amino_acid)\n",
    "\n",
    "# Create a dataframe for amino_acid_variant (A)\n",
    "dataframe_amino_acid_variant = pd.DataFrame(\n",
    "    amino_acid_variant,\n",
    "    columns=list('A'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Generate a random dataframe for nucleotide variant (e.g. c.314C>T). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat \"c.\" for 1000 times to create list\n",
    "c_repeat_1000 = [\"c.\"] * 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate the random amino acid position\n",
    "random_digit_nucleotide_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "# Convert list of interger to list of string\n",
    "random_digit_nucleotide = list()\n",
    "for element in random_digit_nucleotide_int:\n",
    "    random_digit_nucleotide.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for single letter nucleotides (i.e. A, C, G, T) mutation combinations\n",
    "nucleotide_mutation = [\"C>T\", \"T>C\", \"A>G\", \"G>A\"]\n",
    "\n",
    "# Generate the random p.amino_acid\n",
    "random_nucleotide_mutation = [random.choice(nucleotide_mutation)\n",
    "                              for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine \"c.\", position #, and nucleotide_mutation\n",
    "# into one string for each index\n",
    "nucleotide_variant = map(lambda a, b, c: a + b + c,\n",
    "                         c_repeat_1000,\n",
    "                         random_digit_nucleotide,\n",
    "                         random_nucleotide_mutation)\n",
    "\n",
    "# Create a dataframe for nucleotide_variant (N)\n",
    "dataframe_nucleotide_variant = pd.DataFrame(\n",
    "    nucleotide_variant, columns=list('N'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Concatenate all dataframes made in Steps 1-5 into one pandas dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data_hypothetical = pd.concat([dataframe_sample,\n",
    "                                        dataframe_reference_genome,\n",
    "                                        dataframe_gene_genome,\n",
    "                                        dataframe_effect,\n",
    "                                        dataframe_amino_acid_variant,\n",
    "                                        dataframe_nucleotide_variant], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S</th>\n",
       "      <th>R</th>\n",
       "      <th>G</th>\n",
       "      <th>E</th>\n",
       "      <th>A</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>31</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Leu67Pro</td>\n",
       "      <td>c.16C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>46</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His96Ile</td>\n",
       "      <td>c.97A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>96</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ala73Met</td>\n",
       "      <td>c.85T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>17</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Lys31Glx</td>\n",
       "      <td>c.27C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>95</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gln55Val</td>\n",
       "      <td>c.83C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>60</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys49Arg</td>\n",
       "      <td>c.13C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys38Val</td>\n",
       "      <td>c.83T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>45</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Thr67Asn</td>\n",
       "      <td>c.97T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>6</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx90Phe</td>\n",
       "      <td>c.65T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>71</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr31Glu</td>\n",
       "      <td>c.50C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>76</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asx65Thr</td>\n",
       "      <td>c.86A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>84</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ser31Phe</td>\n",
       "      <td>c.1C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>88</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx24Val</td>\n",
       "      <td>c.53T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>49</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Glu16Met</td>\n",
       "      <td>c.10C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>65</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr45Ser</td>\n",
       "      <td>c.96G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>79</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Pro40Val</td>\n",
       "      <td>c.49T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>34</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Phe40Tyr</td>\n",
       "      <td>c.60G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>62</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Glx58Pro</td>\n",
       "      <td>c.29C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>47</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ser76Ala</td>\n",
       "      <td>c.94A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>65</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Pro88Tyr</td>\n",
       "      <td>c.18G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>86</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ile77Leu</td>\n",
       "      <td>c.94T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>4</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Val45Asx</td>\n",
       "      <td>c.25T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>30</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly67Leu</td>\n",
       "      <td>c.7T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>15</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Trp44Pro</td>\n",
       "      <td>c.6C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>84</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Lys74Gly</td>\n",
       "      <td>c.7A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>71</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ser24Gln</td>\n",
       "      <td>c.52T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>90</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Tyr69Asx</td>\n",
       "      <td>c.36G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>87</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Pro5Cys</td>\n",
       "      <td>c.7T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>20</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asp21Asp</td>\n",
       "      <td>c.19C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>59</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro53Asx</td>\n",
       "      <td>c.95T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>91</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Arg79Arg</td>\n",
       "      <td>c.39C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>63</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gln10Pro</td>\n",
       "      <td>c.66T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>48</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gln57Thr</td>\n",
       "      <td>c.19A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>34</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx13Glx</td>\n",
       "      <td>c.50C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>9</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Phe38Phe</td>\n",
       "      <td>c.45T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>47</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Val70Leu</td>\n",
       "      <td>c.41G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>86</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Cys57Cys</td>\n",
       "      <td>c.30T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>20</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Val61Phe</td>\n",
       "      <td>c.13G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>58</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Trp13Thr</td>\n",
       "      <td>c.39T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>42</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly71Lys</td>\n",
       "      <td>c.27T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>39</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Cys26Leu</td>\n",
       "      <td>c.71G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>80</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His9Glx</td>\n",
       "      <td>c.91G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>82</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser71His</td>\n",
       "      <td>c.34G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>1</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro55Trp</td>\n",
       "      <td>c.33G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>5</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Lys60Glu</td>\n",
       "      <td>c.61C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>80</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glx85Asx</td>\n",
       "      <td>c.72G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>75</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn93Ile</td>\n",
       "      <td>c.8T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>7</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly41Glx</td>\n",
       "      <td>c.50G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>39</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Phe16Asx</td>\n",
       "      <td>c.22A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>74</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Glx28Asp</td>\n",
       "      <td>c.38G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>83</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ser71Arg</td>\n",
       "      <td>c.12C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>78</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Tyr39Asp</td>\n",
       "      <td>c.70A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>69</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glx98Met</td>\n",
       "      <td>c.52G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>44</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys60Asn</td>\n",
       "      <td>c.83A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>26</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ile14His</td>\n",
       "      <td>c.91C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>40</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Trp22Asn</td>\n",
       "      <td>c.85T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>8</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro52Ile</td>\n",
       "      <td>c.8G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>28</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Met1Met</td>\n",
       "      <td>c.50G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>4</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser60Arg</td>\n",
       "      <td>c.45A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>75</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asp91Asn</td>\n",
       "      <td>c.57C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      S         R           G                   E           A        N\n",
       "0    31  AF121240       PreS1    missense_variant  p.Leu67Pro  c.16C>T\n",
       "1    46  AB033554       PreS1    missense_variant  p.His96Ile  c.97A>G\n",
       "2    96  AB033554     PreCore  intragenic_variant  p.Ala73Met  c.85T>C\n",
       "3    17  AB033554       PreS2  intragenic_variant  p.Lys31Glx  c.27C>T\n",
       "4    95  AB033556       HBsAg  intragenic_variant  p.Gln55Val  c.83C>T\n",
       "5    60  AB033554       PreS2    missense_variant  p.Lys49Arg  c.13C>T\n",
       "6     7  AF121240       HBsAg    missense_variant  p.Lys38Val  c.83T>C\n",
       "7    45  AF090842           X    missense_variant  p.Thr67Asn  c.97T>C\n",
       "8     6  AB032431     PreCore    missense_variant  p.Asx90Phe  c.65T>C\n",
       "9    71  AF121240       HBsAg    missense_variant  p.Tyr31Glu  c.50C>T\n",
       "10   76  AF121240       PreS2  intragenic_variant  p.Asx65Thr  c.86A>G\n",
       "11   84  AB033554          RT  intragenic_variant  p.Ser31Phe   c.1C>T\n",
       "12   88  AB033554     PreCore    missense_variant  p.Asx24Val  c.53T>C\n",
       "13   49  AB033556     PreCore  intragenic_variant  p.Glu16Met  c.10C>T\n",
       "14   65  AF090842     PreCore    missense_variant  p.Tyr45Ser  c.96G>A\n",
       "15   79  AB033554          RT    missense_variant  p.Pro40Val  c.49T>C\n",
       "16   34  AF121240  Polymerase    missense_variant  p.Phe40Tyr  c.60G>A\n",
       "17   62  AF121240           X  intragenic_variant  p.Glx58Pro  c.29C>T\n",
       "18   47  AB032431          RT  intragenic_variant  p.Ser76Ala  c.94A>G\n",
       "19   65  AF121240       HBsAg    missense_variant  p.Pro88Tyr  c.18G>A\n",
       "20   86  AF121240  Polymerase    missense_variant  p.Ile77Leu  c.94T>C\n",
       "21    4  AF090842       HBsAg    missense_variant  p.Val45Asx  c.25T>C\n",
       "22   30  AB033556           X    missense_variant  p.Gly67Leu   c.7T>C\n",
       "23   15  AB033556       HBsAg  intragenic_variant  p.Trp44Pro   c.6C>T\n",
       "24   84  AB032431          RT  intragenic_variant  p.Lys74Gly   c.7A>G\n",
       "25   71  AF090842          RT  intragenic_variant  p.Ser24Gln  c.52T>C\n",
       "26   90  AB033554  Polymerase  intragenic_variant  p.Tyr69Asx  c.36G>A\n",
       "27   87  AB032431  Polymerase    missense_variant   p.Pro5Cys   c.7T>C\n",
       "28   20  AB032431  Polymerase  intragenic_variant  p.Asp21Asp  c.19C>T\n",
       "29   59  AB032431        Core  intragenic_variant  p.Pro53Asx  c.95T>C\n",
       "..   ..       ...         ...                 ...         ...      ...\n",
       "970  91  AB033556       HBsAg    missense_variant  p.Arg79Arg  c.39C>T\n",
       "971  63  AF090842          RT    missense_variant  p.Gln10Pro  c.66T>C\n",
       "972  48  AF121240       HBsAg  intragenic_variant  p.Gln57Thr  c.19A>G\n",
       "973  34  AF121240     PreCore    missense_variant  p.Asx13Glx  c.50C>T\n",
       "974   9  AF121240           X    missense_variant  p.Phe38Phe  c.45T>C\n",
       "975  47  AB033554     PreCore  intragenic_variant  p.Val70Leu  c.41G>A\n",
       "976  86  AB033556        Core  intragenic_variant  p.Cys57Cys  c.30T>C\n",
       "977  20  AB032431       PreS2    missense_variant  p.Val61Phe  c.13G>A\n",
       "978  58  AB032431          RT  intragenic_variant  p.Trp13Thr  c.39T>C\n",
       "979  42  AB033554          RT    missense_variant  p.Gly71Lys  c.27T>C\n",
       "980  39  AF090842  Polymerase    missense_variant  p.Cys26Leu  c.71G>A\n",
       "981  80  AF090842       PreS1    missense_variant   p.His9Glx  c.91G>A\n",
       "982  82  AF121240       HBsAg    missense_variant  p.Ser71His  c.34G>A\n",
       "983   1  AF121240           X  intragenic_variant  p.Pro55Trp  c.33G>A\n",
       "984   5  AB033556       PreS2  intragenic_variant  p.Lys60Glu  c.61C>T\n",
       "985  80  AB033554           X    missense_variant  p.Glx85Asx  c.72G>A\n",
       "986  75  AF090842     PreCore    missense_variant  p.Asn93Ile   c.8T>C\n",
       "987   7  AB033554       HBsAg    missense_variant  p.Gly41Glx  c.50G>A\n",
       "988  39  AB033554       PreS2    missense_variant  p.Phe16Asx  c.22A>G\n",
       "989  74  AF121240           X  intragenic_variant  p.Glx28Asp  c.38G>A\n",
       "990  83  AB033556           X  intragenic_variant  p.Ser71Arg  c.12C>T\n",
       "991  78  AB033554        Core  intragenic_variant  p.Tyr39Asp  c.70A>G\n",
       "992  69  AB033554  Polymerase    missense_variant  p.Glx98Met  c.52G>A\n",
       "993  44  AF090842        Core    missense_variant  p.Lys60Asn  c.83A>G\n",
       "994  26  AF121240        Core    missense_variant  p.Ile14His  c.91C>T\n",
       "995  40  AB033556       PreS2    missense_variant  p.Trp22Asn  c.85T>C\n",
       "996   8  AB032431       PreS1  intragenic_variant  p.Pro52Ile   c.8G>A\n",
       "997  28  AF121240        Core  intragenic_variant   p.Met1Met  c.50G>A\n",
       "998   4  AB033556           X    missense_variant  p.Ser60Arg  c.45A>G\n",
       "999  75  AB033554       HBsAg    missense_variant  p.Asp91Asn  c.57C>T\n",
       "\n",
       "[1000 rows x 6 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data_hypothetical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 7: Write pandas dataframe in Step 6 into xlsx file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data_hypothetical.to_csv('./clinical_data_hypothetical.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
