{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Hypothetical HBV Clinical Data Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: In order to mimick the process of data processing used in the study, I have generated hypothetical HBV clinical data. The data shown below has no relation to the original clinical data. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Generate a random dataframe for Sample (i.e. unique patients' number used during the clinical study). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the random dataframe for Sample (S)\n",
    "dataframe_sample = pd.DataFrame(np.random.randint(1,\n",
    "                                                  100, size=(1000, 1)),\n",
    "                                columns=list('S'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Generate a random dataframe for Reference Genome (i.e. genotype-specific reference sequence (GenBank accession numbers: AF090842, AB033554, AB033556, AF121240, and AB032431, for genotypes A to E, respectively))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Reference Genome used in the clinical study\n",
    "GenBank_sequences = [\"AF090842\",\n",
    "                     \"AB033554\",\n",
    "                     \"AB033556\",\n",
    "                     \"AF121240\",\n",
    "                     \"AB032431\"]\n",
    "\n",
    "# Create the dataframe for Reference Genome (R)\n",
    "dataframe_reference_genome = pd.DataFrame(np.random.choice\n",
    "                                          (GenBank_sequences,\n",
    "                                           size=1000,\n",
    "                                           replace=True,\n",
    "                                           p=None),\n",
    "                                          columns=list('R'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 3: Generate a random dataframe for Gene (i.e. genes encoded by the genome: C (PreCore, Core), P (Polymerase, RT), S (PreS1, PreS2, HBsAg), X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Genes used in the clinical study\n",
    "gene_genome = [\"PreCore\", \"Core\",\n",
    "               \"Polymerase\", \"RT\",\n",
    "               \"PreS1\", \"PreS2\",\n",
    "               \"HBsAg\", \"X\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create the dataframe for Genes (G)\n",
    "dataframe_gene_genome = pd.DataFrame(np.random.choice\n",
    "                                     (gene_genome, size=1000,\n",
    "                                      replace=True, p=None),\n",
    "                                     columns=list('G'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 4: Generate a random dataframe for Effect (i.e. intragenic or missense variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List of elements for Effect used in the clinical study\n",
    "effect_variant = [\"intragenic_variant\", \"missense_variant\"]\n",
    "\n",
    "# Create the dataframe for Effect (E)\n",
    "dataframe_effect = pd.DataFrame(np.random.choice(effect_variant,\n",
    "                                                 size=1000,\n",
    "                                                 replace=True,\n",
    "                                                 p=None), columns=list('E'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 5: Generate a random dataframe for amino acid variant (e.g. p.Pro156Ser). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for p.amino_acid (e.g. p.Pro in p.Pro156Ser)\n",
    "p_amino_acid = [\"p.Ala\", \"p.Arg\", \"p.Asn\", \"p.Asp\", \"p.Cys\",\n",
    "                \"p.Gln\", \"p.Glu\", \"p.Gly\", \"p.His\", \"p.Ile\", \"p.Leu\",\n",
    "                \"p.Lys\", \"p.Met\", \"p.Phe\", \"p.Pro\", \"p.Ser\",\n",
    "                \"p.Thr\", \"p.Trp\", \"p.Tyr\", \"p.Val\", \"p.Asx\", \"p.Glx\"]\n",
    "\n",
    "# Generate the random p.amino_acid\n",
    "random_p_amino_acid = [random.choice(p_amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate the random amino acid position\n",
    "random_digit_amino_acid_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "# Convert list of interger to list of string\n",
    "random_digit_amino_acid = list()\n",
    "for element in random_digit_amino_acid_int:\n",
    "    random_digit_amino_acid.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for amino_acid (e.g. Ser in p.Pro156Ser)\n",
    "amino_acid = [\"Ala\", \"Arg\", \"Asn\", \"Asp\", \"Cys\",\n",
    "              \"Gln\", \"Glu\", \"Gly\", \"His\", \"Ile\", \"Leu\",\n",
    "              \"Lys\", \"Met\", \"Phe\", \"Pro\", \"Ser\",\n",
    "              \"Thr\", \"Trp\", \"Tyr\", \"Val\", \"Asx\", \"Glx\"]\n",
    "\n",
    "# Generate the random amino_acid\n",
    "random_amino_acid = [random.choice(amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine p.amino_acid, position #\n",
    "# and amino_acid into one string for each index\n",
    "amino_acid_variant = map(lambda a, b, c: a + b + c,\n",
    "                         random_p_amino_acid,\n",
    "                         random_digit_amino_acid,\n",
    "                         random_amino_acid)\n",
    "\n",
    "# Create a dataframe for amino_acid_variant (A)\n",
    "dataframe_amino_acid_variant = pd.DataFrame(\n",
    "    amino_acid_variant,\n",
    "    columns=list('A'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Generate a random dataframe for nucleotide variant (e.g. c.314C>T). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Repeat \"c.\" for 1000 times to create list\n",
    "c_repeat_1000 = [\"c.\"] * 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Generate the random amino acid position\n",
    "random_digit_nucleotide_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "# Convert list of interger to list of string\n",
    "random_digit_nucleotide = list()\n",
    "for element in random_digit_nucleotide_int:\n",
    "    random_digit_nucleotide.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# List for single letter nucleotides (i.e. A, C, G, T) mutation combinations\n",
    "nucleotide_mutation = [\"C>T\", \"T>C\", \"A>G\", \"G>A\"]\n",
    "\n",
    "# Generate the random p.amino_acid\n",
    "random_nucleotide_mutation = [random.choice(nucleotide_mutation)\n",
    "                              for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Combine \"c.\", position #, and nucleotide_mutation\n",
    "# into one string for each index\n",
    "nucleotide_variant = map(lambda a, b, c: a + b + c,\n",
    "                         c_repeat_1000,\n",
    "                         random_digit_nucleotide,\n",
    "                         random_nucleotide_mutation)\n",
    "\n",
    "# Create a dataframe for nucleotide_variant (N)\n",
    "dataframe_nucleotide_variant = pd.DataFrame(\n",
    "    nucleotide_variant, columns=list('N'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Concatenate all dataframes made in Steps 1-5 into one pandas dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data_hypothetical = pd.concat([dataframe_sample,\n",
    "                                        dataframe_reference_genome,\n",
    "                                        dataframe_gene_genome,\n",
    "                                        dataframe_effect,\n",
    "                                        dataframe_amino_acid_variant,\n",
    "                                        dataframe_nucleotide_variant], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S</th>\n",
       "      <th>R</th>\n",
       "      <th>G</th>\n",
       "      <th>E</th>\n",
       "      <th>A</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>48</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His8Glu</td>\n",
       "      <td>c.90A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>25</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu99Glx</td>\n",
       "      <td>c.70A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>90</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu97Ile</td>\n",
       "      <td>c.36A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>67</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asn58Lys</td>\n",
       "      <td>c.51T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser84His</td>\n",
       "      <td>c.69T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>87</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr14Ser</td>\n",
       "      <td>c.37A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>97</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gln67Thr</td>\n",
       "      <td>c.65A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Arg88Val</td>\n",
       "      <td>c.54G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>41</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx59Pro</td>\n",
       "      <td>c.52T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>93</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro32Gln</td>\n",
       "      <td>c.7A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>79</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asp65Gly</td>\n",
       "      <td>c.64T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>79</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Cys92Asx</td>\n",
       "      <td>c.69A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>70</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ala13Lys</td>\n",
       "      <td>c.88T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>27</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe88Ile</td>\n",
       "      <td>c.33A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>9</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Phe62Met</td>\n",
       "      <td>c.29T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>13</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.His58Asp</td>\n",
       "      <td>c.50A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>82</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asp31Val</td>\n",
       "      <td>c.80C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>9</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser44Trp</td>\n",
       "      <td>c.43A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>37</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser94Asn</td>\n",
       "      <td>c.28T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>35</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ala1Ala</td>\n",
       "      <td>c.82G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>13</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn66Ser</td>\n",
       "      <td>c.99C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>72</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly93Glu</td>\n",
       "      <td>c.82G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>74</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asp84Pro</td>\n",
       "      <td>c.13C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>48</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His76Trp</td>\n",
       "      <td>c.90A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gln25Trp</td>\n",
       "      <td>c.91G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asx90His</td>\n",
       "      <td>c.12C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>54</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Arg71Thr</td>\n",
       "      <td>c.82C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>92</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe45Asx</td>\n",
       "      <td>c.88T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>94</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro54Lys</td>\n",
       "      <td>c.78G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>78</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Trp6Phe</td>\n",
       "      <td>c.49A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>59</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ala45Ser</td>\n",
       "      <td>c.25A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>94</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asn29Gln</td>\n",
       "      <td>c.45T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>47</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro63His</td>\n",
       "      <td>c.70A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>52</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Cys57His</td>\n",
       "      <td>c.12T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>58</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu75Asp</td>\n",
       "      <td>c.29A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>46</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser68Val</td>\n",
       "      <td>c.49C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>61</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr35Asx</td>\n",
       "      <td>c.23A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>10</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly24Asn</td>\n",
       "      <td>c.92A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>79</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Thr54Asx</td>\n",
       "      <td>c.37G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>39</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His43Gly</td>\n",
       "      <td>c.55G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>83</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro7Glu</td>\n",
       "      <td>c.69G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>31</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asx59Gly</td>\n",
       "      <td>c.36C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>66</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser14His</td>\n",
       "      <td>c.84C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>21</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gln71Val</td>\n",
       "      <td>c.44G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>9</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Arg64Gln</td>\n",
       "      <td>c.62C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>56</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ala92Tyr</td>\n",
       "      <td>c.74T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>66</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Thr19Val</td>\n",
       "      <td>c.90A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>23</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.His99Cys</td>\n",
       "      <td>c.57G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>5</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Pro37His</td>\n",
       "      <td>c.23A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>8</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ala10Ile</td>\n",
       "      <td>c.54C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>24</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Phe46Asp</td>\n",
       "      <td>c.75A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>93</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Cys78Val</td>\n",
       "      <td>c.93C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>93</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.His58His</td>\n",
       "      <td>c.97A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>51</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ser47His</td>\n",
       "      <td>c.35A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>76</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Arg64Trp</td>\n",
       "      <td>c.42A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>14</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu26Gln</td>\n",
       "      <td>c.32G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>52</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.His65Trp</td>\n",
       "      <td>c.39C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>1</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr61His</td>\n",
       "      <td>c.45G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>29</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>RT</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe31Arg</td>\n",
       "      <td>c.66T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>15</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glx68His</td>\n",
       "      <td>c.52T&gt;C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      S         R           G                   E           A        N\n",
       "0    48  AF090842     PreCore    missense_variant   p.His8Glu  c.90A>G\n",
       "1    25  AB032431        Core  intragenic_variant  p.Leu99Glx  c.70A>G\n",
       "2    90  AB032431        Core  intragenic_variant  p.Leu97Ile  c.36A>G\n",
       "3    67  AB033554       PreS1  intragenic_variant  p.Asn58Lys  c.51T>C\n",
       "4    15  AB032431          RT    missense_variant  p.Ser84His  c.69T>C\n",
       "5    87  AB033554       HBsAg    missense_variant  p.Tyr14Ser  c.37A>G\n",
       "6    97  AB033554       HBsAg  intragenic_variant  p.Gln67Thr  c.65A>G\n",
       "7     6  AB033554     PreCore  intragenic_variant  p.Arg88Val  c.54G>A\n",
       "8    41  AF090842          RT    missense_variant  p.Asx59Pro  c.52T>C\n",
       "9    93  AF121240       HBsAg  intragenic_variant  p.Pro32Gln   c.7A>G\n",
       "10   79  AF121240       PreS1    missense_variant  p.Asp65Gly  c.64T>C\n",
       "11   79  AF090842           X    missense_variant  p.Cys92Asx  c.69A>G\n",
       "12   70  AB033554          RT    missense_variant  p.Ala13Lys  c.88T>C\n",
       "13   27  AB033554           X  intragenic_variant  p.Phe88Ile  c.33A>G\n",
       "14    9  AB032431        Core    missense_variant  p.Phe62Met  c.29T>C\n",
       "15   13  AF090842       PreS2  intragenic_variant  p.His58Asp  c.50A>G\n",
       "16   82  AB033554     PreCore  intragenic_variant  p.Asp31Val  c.80C>T\n",
       "17    9  AB032431        Core    missense_variant  p.Ser44Trp  c.43A>G\n",
       "18   37  AB033556     PreCore    missense_variant  p.Ser94Asn  c.28T>C\n",
       "19   35  AB033556          RT    missense_variant   p.Ala1Ala  c.82G>A\n",
       "20   13  AB033554       PreS2    missense_variant  p.Asn66Ser  c.99C>T\n",
       "21   72  AB032431  Polymerase    missense_variant  p.Gly93Glu  c.82G>A\n",
       "22   74  AF121240       PreS2  intragenic_variant  p.Asp84Pro  c.13C>T\n",
       "23   48  AB033556           X    missense_variant  p.His76Trp  c.90A>G\n",
       "24   25  AB033556          RT  intragenic_variant  p.Gln25Trp  c.91G>A\n",
       "25   26  AB032431        Core  intragenic_variant  p.Asx90His  c.12C>T\n",
       "26   54  AF090842          RT    missense_variant  p.Arg71Thr  c.82C>T\n",
       "27   92  AB033554     PreCore  intragenic_variant  p.Phe45Asx  c.88T>C\n",
       "28   94  AF090842     PreCore  intragenic_variant  p.Pro54Lys  c.78G>A\n",
       "29   78  AF121240       HBsAg    missense_variant   p.Trp6Phe  c.49A>G\n",
       "..   ..       ...         ...                 ...         ...      ...\n",
       "970  59  AF121240           X    missense_variant  p.Ala45Ser  c.25A>G\n",
       "971  94  AF090842     PreCore  intragenic_variant  p.Asn29Gln  c.45T>C\n",
       "972  47  AF090842  Polymerase  intragenic_variant  p.Pro63His  c.70A>G\n",
       "973  52  AB033556       HBsAg  intragenic_variant  p.Cys57His  c.12T>C\n",
       "974  58  AF121240       PreS1  intragenic_variant  p.Leu75Asp  c.29A>G\n",
       "975  46  AF121240        Core    missense_variant  p.Ser68Val  c.49C>T\n",
       "976  61  AF121240        Core    missense_variant  p.Tyr35Asx  c.23A>G\n",
       "977  10  AB033554     PreCore    missense_variant  p.Gly24Asn  c.92A>G\n",
       "978  79  AF090842          RT    missense_variant  p.Thr54Asx  c.37G>A\n",
       "979  39  AB033556       PreS2    missense_variant  p.His43Gly  c.55G>A\n",
       "980  83  AF121240       PreS2  intragenic_variant   p.Pro7Glu  c.69G>A\n",
       "981  31  AB033556       PreS1  intragenic_variant  p.Asx59Gly  c.36C>T\n",
       "982  66  AF121240     PreCore    missense_variant  p.Ser14His  c.84C>T\n",
       "983  21  AB033556        Core    missense_variant  p.Gln71Val  c.44G>A\n",
       "984   9  AB033554          RT  intragenic_variant  p.Arg64Gln  c.62C>T\n",
       "985  56  AB032431  Polymerase  intragenic_variant  p.Ala92Tyr  c.74T>C\n",
       "986  66  AF090842          RT  intragenic_variant  p.Thr19Val  c.90A>G\n",
       "987  23  AF090842  Polymerase  intragenic_variant  p.His99Cys  c.57G>A\n",
       "988   5  AB033556       PreS1    missense_variant  p.Pro37His  c.23A>G\n",
       "989   8  AB033556       PreS2    missense_variant  p.Ala10Ile  c.54C>T\n",
       "990  24  AB033556       HBsAg    missense_variant  p.Phe46Asp  c.75A>G\n",
       "991  93  AB032431           X  intragenic_variant  p.Cys78Val  c.93C>T\n",
       "992  93  AB033556  Polymerase  intragenic_variant  p.His58His  c.97A>G\n",
       "993  51  AF090842       HBsAg  intragenic_variant  p.Ser47His  c.35A>G\n",
       "994  76  AB033554        Core    missense_variant  p.Arg64Trp  c.42A>G\n",
       "995  14  AB033554       PreS1  intragenic_variant  p.Leu26Gln  c.32G>A\n",
       "996  52  AB033556  Polymerase  intragenic_variant  p.His65Trp  c.39C>T\n",
       "997   1  AB032431       PreS1    missense_variant  p.Tyr61His  c.45G>A\n",
       "998  29  AB033554          RT  intragenic_variant  p.Phe31Arg  c.66T>C\n",
       "999  15  AB033556           X    missense_variant  p.Glx68His  c.52T>C\n",
       "\n",
       "[1000 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data_hypothetical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 7: Write pandas dataframe in Step 6 into xlsx file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data_hypothetical.to_csv('./clinical_data_hypothetical.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
