{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Hypothetical HBV Clinical Data Generation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: In order to mimick the process of data processing used in the study, I have generated hypothetical HBV clinical data. The data shown below has no relation to the original clinical data. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import random"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Generate a random dataframe for Sample (i.e. unique patients' number used during the clinical study). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create the random dataframe for Sample (S)\n",
    "dataframe_sample = pd.DataFrame(np.random.randint(1,100,size=(1000, 1)), columns=list('S'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Generate a random dataframe for Reference Genome (i.e. genotype-specific reference sequence (GenBank accession numbers: AF090842, AB033554, AB033556, AF121240, and AB032431, for genotypes A to E, respectively))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List of elements for Reference Genome used in the clinical study \n",
    "GenBank_sequences = [\"AF090842\", \"AB033554\", \"AB033556\", \"AF121240\", \"AB032431\"]\n",
    "\n",
    "#Create the dataframe for Reference Genome (R)\n",
    "dataframe_reference_genome = pd.DataFrame(np.random.choice(GenBank_sequences, size=1000, replace=True, p=None), columns=list('R'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 3: Generate a random dataframe for Gene (i.e. genes encoded by the genome: C (PreCore, Core), P (Polymerase, RT), S (PreS1, PreS2, HBsAg), X)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List of elements for Genes used in the clinical study \n",
    "gene_genome = [\"PreCore\", \"Core\", \"Polymerase\", \"RT\", \"PreS1\", \"PreS2\", \"HBsAg\", \"X\"]\n",
    "\n",
    "#Create the dataframe for Genes (G)\n",
    "dataframe_gene_genome = pd.DataFrame(np.random.choice(gene_genome, size=1000, replace=True, p=None), columns=list('G'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 4: Generate a random dataframe for Effect (i.e. intragenic or missense variant)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List of elements for Effect used in the clinical study \n",
    "effect_variant = [\"intragenic_variant\", \"missense_variant\"]\n",
    "\n",
    "#Create the dataframe for Effect (E)\n",
    "dataframe_effect = pd.DataFrame(np.random.choice(effect_variant, size=1000, replace=True, p=None), columns=list('E'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 5: Generate a random dataframe for amino acid variant (e.g. p.Pro156Ser). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List for p.amino_acid (e.g. p.Pro in p.Pro156Ser)\n",
    "p_amino_acid = [\"p.Ala\", \"p.Arg\", \"p.Asn\", \"p.Asp\", \"p.Cys\", \"p.Gln\", \"p.Glu\", \"p.Gly\", \"p.His\", \"p.Ile\", \"p.Leu\", \"p.Lys\", \"p.Met\", \"p.Phe\", \"p.Pro\", \"p.Ser\", \"p.Thr\", \"p.Trp\", \"p.Tyr\", \"p.Val\", \"p.Asx\", \"p.Glx\"]\n",
    "\n",
    "#Generate the random p.amino_acid \n",
    "random_p_amino_acid = [random.choice(p_amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Generate the random amino acid position \n",
    "random_digit_amino_acid_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "#Convert list of interger to list of string \n",
    "random_digit_amino_acid = list()\n",
    "for element in random_digit_amino_acid_int:\n",
    "    random_digit_amino_acid.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List for amino_acid (e.g. Ser in p.Pro156Ser)\n",
    "amino_acid = [\"Ala\", \"Arg\", \"Asn\", \"Asp\", \"Cys\", \"Gln\", \"Glu\", \"Gly\", \"His\", \"Ile\", \"Leu\", \"Lys\", \"Met\", \"Phe\", \"Pro\", \"Ser\", \"Thr\", \"Trp\", \"Tyr\", \"Val\", \"Asx\", \"Glx\"]\n",
    "\n",
    "#Generate the random amino_acid\n",
    "random_amino_acid = [random.choice(amino_acid) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Combine p.amino_acid, position #, and amino_acid into one string for each index\n",
    "amino_acid_variant = map(lambda a, b, c: a + b + c, random_p_amino_acid, random_digit_amino_acid, random_amino_acid)\n",
    "\n",
    "#Create a dataframe for amino_acid_variant (A)\n",
    "dataframe_amino_acid_variant = pd.DataFrame(amino_acid_variant, columns=list('A'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Generate a random dataframe for nucleotide variant (e.g. c.314C>T). "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Repeat \"c.\" for 1000 times to create list\n",
    "c_repeat_1000 = [\"c.\"] * 1000"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Generate the random amino acid position \n",
    "random_digit_nucleotide_int = [int(100*random.random()) for i in range(1000)]\n",
    "\n",
    "#Convert list of interger to list of string \n",
    "random_digit_nucleotide = list()\n",
    "for element in random_digit_nucleotide_int:\n",
    "    random_digit_nucleotide.append(str(element))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [],
   "source": [
    "#List for single letter nucleotides (i.e. A, C, G, T) mutation combinations \n",
    "nucleotide_mutation = [\"C>T\", \"T>C\", \"A>G\", \"G>A\"]\n",
    "\n",
    "#Generate the random p.amino_acid \n",
    "random_nucleotide_mutation = [random.choice(nucleotide_mutation) for i in range(1000)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Combine \"c.\", position #, and nucleotide_mutation into one string for each index\n",
    "nucleotide_variant = map(lambda a, b, c: a + b + c, c_repeat_1000, random_digit_nucleotide, random_nucleotide_mutation)\n",
    "\n",
    "#Create a dataframe for nucleotide_variant (N)\n",
    "dataframe_nucleotide_variant = pd.DataFrame(nucleotide_variant, columns=list('N'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 6: Concatenate all dataframes made in Steps 1-5 into one pandas dataframe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinical_data_hypothetical = pd.concat([dataframe_sample, dataframe_reference_genome, dataframe_gene_genome, dataframe_effect, dataframe_amino_acid_variant, dataframe_nucleotide_variant], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S</th>\n",
       "      <th>R</th>\n",
       "      <th>G</th>\n",
       "      <th>E</th>\n",
       "      <th>A</th>\n",
       "      <th>N</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>82</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn25Gly</td>\n",
       "      <td>c.71G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>69</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx81Cys</td>\n",
       "      <td>c.96A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>38</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glu93Leu</td>\n",
       "      <td>c.26T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>41</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Arg8Phe</td>\n",
       "      <td>c.87A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Arg12Phe</td>\n",
       "      <td>c.82A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>89</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Val32Glu</td>\n",
       "      <td>c.5T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>51</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Tyr86Trp</td>\n",
       "      <td>c.29T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>63</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys39Glu</td>\n",
       "      <td>c.27G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>27</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Ser86Gly</td>\n",
       "      <td>c.58T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>76</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn68Glx</td>\n",
       "      <td>c.51C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>35</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Trp30Tyr</td>\n",
       "      <td>c.97G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>97</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu38Asn</td>\n",
       "      <td>c.7T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>68</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe11Val</td>\n",
       "      <td>c.53C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>69</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Glx16Ser</td>\n",
       "      <td>c.56T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>23</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Cys14Phe</td>\n",
       "      <td>c.0C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>93</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe6Glx</td>\n",
       "      <td>c.52A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>45</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Core</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ala24Tyr</td>\n",
       "      <td>c.53T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>86</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys18Lys</td>\n",
       "      <td>c.73G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>93</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gly71Leu</td>\n",
       "      <td>c.19T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>42</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glu37Trp</td>\n",
       "      <td>c.88A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>33</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His26Trp</td>\n",
       "      <td>c.86C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>51</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Met18Glx</td>\n",
       "      <td>c.21A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>25</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr56Pro</td>\n",
       "      <td>c.80G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>72</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx9Ile</td>\n",
       "      <td>c.24G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>99</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>HBsAg</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn24Phe</td>\n",
       "      <td>c.3T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>47</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gly61Val</td>\n",
       "      <td>c.88G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>53</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Tyr32His</td>\n",
       "      <td>c.8T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>57</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Met70Phe</td>\n",
       "      <td>c.85G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>88</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Trp73Ala</td>\n",
       "      <td>c.41G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>25</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asn16Thr</td>\n",
       "      <td>c.70A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>66</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Leu29Tyr</td>\n",
       "      <td>c.37G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>26</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Arg56Met</td>\n",
       "      <td>c.76G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>54</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Thr13Gln</td>\n",
       "      <td>c.16G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>90</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr0Glu</td>\n",
       "      <td>c.76C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>41</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Thr92Met</td>\n",
       "      <td>c.15C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>96</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Lys19Asn</td>\n",
       "      <td>c.48G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>87</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Asx12Arg</td>\n",
       "      <td>c.49T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>56</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu83Ser</td>\n",
       "      <td>c.54C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>44</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreS2</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Phe75Phe</td>\n",
       "      <td>c.23A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>20</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gly36Glu</td>\n",
       "      <td>c.49T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>25</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr43Arg</td>\n",
       "      <td>c.45C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>46</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr89Asn</td>\n",
       "      <td>c.84A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>68</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gly81Ala</td>\n",
       "      <td>c.57C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>95</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Asx32Glu</td>\n",
       "      <td>c.18G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>24</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>X</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Met23Ser</td>\n",
       "      <td>c.43T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>50</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreS1</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Cys70Glx</td>\n",
       "      <td>c.88A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>1</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Gly52Gln</td>\n",
       "      <td>c.22T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>84</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.His35Cys</td>\n",
       "      <td>c.69T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>988</th>\n",
       "      <td>43</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Gly17Glx</td>\n",
       "      <td>c.13A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989</th>\n",
       "      <td>60</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Glu74Asp</td>\n",
       "      <td>c.48G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>42</td>\n",
       "      <td>AF090842</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Leu0Arg</td>\n",
       "      <td>c.11G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>45</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Arg69Phe</td>\n",
       "      <td>c.9T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>39</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Tyr27Phe</td>\n",
       "      <td>c.56T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>993</th>\n",
       "      <td>44</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Ala69Ser</td>\n",
       "      <td>c.89G&gt;A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>994</th>\n",
       "      <td>66</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Pro48Gly</td>\n",
       "      <td>c.37C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>995</th>\n",
       "      <td>77</td>\n",
       "      <td>AB033556</td>\n",
       "      <td>X</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Cys58Asp</td>\n",
       "      <td>c.10C&gt;T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>996</th>\n",
       "      <td>84</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>PreCore</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>p.Leu43Arg</td>\n",
       "      <td>c.49A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>997</th>\n",
       "      <td>12</td>\n",
       "      <td>AB033554</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Lys9Arg</td>\n",
       "      <td>c.99A&gt;G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>998</th>\n",
       "      <td>27</td>\n",
       "      <td>AB032431</td>\n",
       "      <td>Polymerase</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Tyr22His</td>\n",
       "      <td>c.66T&gt;C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>999</th>\n",
       "      <td>22</td>\n",
       "      <td>AF121240</td>\n",
       "      <td>Core</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>p.Arg73Lys</td>\n",
       "      <td>c.1C&gt;T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      S         R           G                   E           A        N\n",
       "0    82  AB033556        Core    missense_variant  p.Asn25Gly  c.71G>A\n",
       "1    69  AF090842  Polymerase    missense_variant  p.Asx81Cys  c.96A>G\n",
       "2    38  AB033556       HBsAg    missense_variant  p.Glu93Leu  c.26T>C\n",
       "3    41  AB033554       PreS2    missense_variant   p.Arg8Phe  c.87A>G\n",
       "4     9  AB033556       PreS2  intragenic_variant  p.Arg12Phe  c.82A>G\n",
       "5    89  AB032431  Polymerase  intragenic_variant  p.Val32Glu   c.5T>C\n",
       "6    51  AF090842     PreCore  intragenic_variant  p.Tyr86Trp  c.29T>C\n",
       "7    63  AB033556       PreS2    missense_variant  p.Lys39Glu  c.27G>A\n",
       "8    27  AB033554       PreS1    missense_variant  p.Ser86Gly  c.58T>C\n",
       "9    76  AB032431       PreS1    missense_variant  p.Asn68Glx  c.51C>T\n",
       "10   35  AF121240           X    missense_variant  p.Trp30Tyr  c.97G>A\n",
       "11   97  AB033554        Core  intragenic_variant  p.Leu38Asn   c.7T>C\n",
       "12   68  AF090842  Polymerase  intragenic_variant  p.Phe11Val  c.53C>T\n",
       "13   69  AB033556     PreCore  intragenic_variant  p.Glx16Ser  c.56T>C\n",
       "14   23  AB032431       HBsAg    missense_variant  p.Cys14Phe   c.0C>T\n",
       "15   93  AB033554     PreCore  intragenic_variant   p.Phe6Glx  c.52A>G\n",
       "16   45  AB032431        Core  intragenic_variant  p.Ala24Tyr  c.53T>C\n",
       "17   86  AB032431       PreS1    missense_variant  p.Lys18Lys  c.73G>A\n",
       "18   93  AF090842  Polymerase  intragenic_variant  p.Gly71Leu  c.19T>C\n",
       "19   42  AF121240       PreS1    missense_variant  p.Glu37Trp  c.88A>G\n",
       "20   33  AF090842       PreS1    missense_variant  p.His26Trp  c.86C>T\n",
       "21   51  AB033554           X    missense_variant  p.Met18Glx  c.21A>G\n",
       "22   25  AB032431  Polymerase    missense_variant  p.Tyr56Pro  c.80G>A\n",
       "23   72  AF090842       PreS2    missense_variant   p.Asx9Ile  c.24G>A\n",
       "24   99  AF090842       HBsAg    missense_variant  p.Asn24Phe   c.3T>C\n",
       "25   47  AB032431     PreCore  intragenic_variant  p.Gly61Val  c.88G>A\n",
       "26   53  AF121240     PreCore  intragenic_variant  p.Tyr32His   c.8T>C\n",
       "27   57  AB032431           X  intragenic_variant  p.Met70Phe  c.85G>A\n",
       "28   88  AB033554       PreS2  intragenic_variant  p.Trp73Ala  c.41G>A\n",
       "29   25  AF121240        Core    missense_variant  p.Asn16Thr  c.70A>G\n",
       "..   ..       ...         ...                 ...         ...      ...\n",
       "970  66  AF121240           X    missense_variant  p.Leu29Tyr  c.37G>A\n",
       "971  26  AF090842       PreS1  intragenic_variant  p.Arg56Met  c.76G>A\n",
       "972  54  AF121240     PreCore  intragenic_variant  p.Thr13Gln  c.16G>A\n",
       "973  90  AB033556     PreCore    missense_variant   p.Tyr0Glu  c.76C>T\n",
       "974  41  AF090842  Polymerase    missense_variant  p.Thr92Met  c.15C>T\n",
       "975  96  AF090842           X  intragenic_variant  p.Lys19Asn  c.48G>A\n",
       "976  87  AB033554  Polymerase  intragenic_variant  p.Asx12Arg  c.49T>C\n",
       "977  56  AB033556  Polymerase  intragenic_variant  p.Leu83Ser  c.54C>T\n",
       "978  44  AB033556       PreS2  intragenic_variant  p.Phe75Phe  c.23A>G\n",
       "979  20  AB032431     PreCore  intragenic_variant  p.Gly36Glu  c.49T>C\n",
       "980  25  AB033556     PreCore    missense_variant  p.Tyr43Arg  c.45C>T\n",
       "981  46  AF121240        Core    missense_variant  p.Tyr89Asn  c.84A>G\n",
       "982  68  AF090842       PreS1  intragenic_variant  p.Gly81Ala  c.57C>T\n",
       "983  95  AF090842        Core    missense_variant  p.Asx32Glu  c.18G>A\n",
       "984  24  AB033554           X    missense_variant  p.Met23Ser  c.43T>C\n",
       "985  50  AF121240       PreS1  intragenic_variant  p.Cys70Glx  c.88A>G\n",
       "986   1  AF121240     PreCore    missense_variant  p.Gly52Gln  c.22T>C\n",
       "987  84  AB033556        Core    missense_variant  p.His35Cys  c.69T>C\n",
       "988  43  AF090842     PreCore  intragenic_variant  p.Gly17Glx  c.13A>G\n",
       "989  60  AF121240     PreCore    missense_variant  p.Glu74Asp  c.48G>A\n",
       "990  42  AF090842        Core    missense_variant   p.Leu0Arg  c.11G>A\n",
       "991  45  AB033554  Polymerase  intragenic_variant  p.Arg69Phe   c.9T>C\n",
       "992  39  AB032431     PreCore  intragenic_variant  p.Tyr27Phe  c.56T>C\n",
       "993  44  AF121240  Polymerase  intragenic_variant  p.Ala69Ser  c.89G>A\n",
       "994  66  AB032431     PreCore  intragenic_variant  p.Pro48Gly  c.37C>T\n",
       "995  77  AB033556           X  intragenic_variant  p.Cys58Asp  c.10C>T\n",
       "996  84  AB032431     PreCore  intragenic_variant  p.Leu43Arg  c.49A>G\n",
       "997  12  AB033554  Polymerase    missense_variant   p.Lys9Arg  c.99A>G\n",
       "998  27  AB032431  Polymerase    missense_variant  p.Tyr22His  c.66T>C\n",
       "999  22  AF121240        Core    missense_variant  p.Arg73Lys   c.1C>T\n",
       "\n",
       "[1000 rows x 6 columns]"
      ]
     },
     "execution_count": 197,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical_data_hypothetical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 7: Write pandas dataframe in Step 6 into xlsx file "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [],
   "source": [
    "#clinical_data_hypothetical.to_csv('./clinical_data_hypothetical.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
